src/patchkernel/patch_kernel.cpp

```diff
@@ -5635,6 +5635,147 @@ void PatchKernel::getInterfaceVertexCoordinates(long id, std::array<double, 3> *
+                                          std::vector<std::array<double, 3>> *polygon) const
+{
+        double distanceTolerance = getTol();
+
+	const Interface &interface = getInterface(id);
+	ElementType elementType = interface.getType();
+	switch (elementType)
+	{
+
+	case ElementType::LINE:
+	{
+		const std::array<double, 3> &P0 = getVertexCoords(interface.getVertexId(0));
+		const std::array<double, 3> &P1 = getVertexCoords(interface.getVertexId(1));
+
+		std::array<double, 3> x;
+		bool intrExist = CGElem::intersectSegmentPlane(P0, P1, P, nP, x, distanceTolerance);
+
+		if (norm2(x - P0) < distanceTolerance) return false;
+		if (norm2(x - P1) < distanceTolerance) return false;
+
+		if (!intrExist) return false;
+
+		(*intersection)[0] = x;
+		(*intersection)[1] = x;
+                if (dotProduct(P - P0, nP) > 0.) {
+			polygon->push_back(P0);
+			polygon->push_back(x);
+                        return true;
+                }
+		polygon->push_back(x);
+		polygon->push_back(P1);
+                return true;
+	}
+
+	case ElementType::TRIANGLE:
+	{
+		const std::array<double, 3> &P0 = getVertexCoords(interface.getVertexId(0));
+		const std::array<double, 3> &P1 = getVertexCoords(interface.getVertexId(1));
+		const std::array<double, 3> &P2 = getVertexCoords(interface.getVertexId(2));
+            
+		return CGElem::intersectPlaneTriangle(P0, P1, P2, P, nP,
+                       *intersection, *polygon, distanceTolerance);
+	}
+
+	case ElementType::PIXEL:
+	{
+		// evaluate pixel normal
+		std::unique_ptr<std::array<double, 3>[]> coordinates;
+		getElementVertexCoordinates(interface, &coordinates);
+		std::array<double, 3> normal = interface.evalNormal(coordinates.get());
+
+		// evaluate the minimum and maximum coordinates of the pixel
+		std::array<double,3> minPoint;
+		std::array<double,3> maxPoint;
+		evalElementBoundingBox(interface, &minPoint, &maxPoint);
+
+		// evaluate direction of pixel
+		int dir = -1;
+                double maxCoor = 0.;
+		for (int i = 0; i < 3; ++i) {
+			if (std::abs(normal[i]) > 0.) {
+				maxCoor = normal[i];
+				dir     = i;
+			}
+		}
+
+		// evaluate the constant coordinate over the pixel
+		double constCoor = minPoint[dir];
+
+		// evaluate the directions tangential to the pixel plane
+		int d0 = (dir + 1) % 3;
+		int d1 = (dir + 2) % 3;
+
+		// evaluate the direction's sign
+		++dir;
+		if (maxCoor < 0.) {
+			dir = -dir;
+                }
+
+		// evaluate the minimum and maximum coordinates in
+		// the pixel's coordinate system
+		std::array<double, 2> A0 = {minPoint[d0], minPoint[d1]};
+		std::array<double, 2> A1 = {maxPoint[d0], maxPoint[d1]};
+
+		return CGElem::intersectPlanePixel(P, nP, A0, A1, dir, constCoor,
+                       *intersection, *polygon, distanceTolerance);
+	}
+
+	default:
+	{
+	        ConstProxyVector<long> vertexIds = interface.getVertexIds();
+                int nVertices = vertexIds.size();
```

Comment:
You should get the vertices in counter clockwise order:
```
assert(interfaceType != ElementType::UNDEFINED);
const Reference2DElementInfo &interfaceInfo = static_cast<const Reference2DElementInfo &>(interfaceType);

std::vector<std::array<double, 3>> coordinates(nVertices)
for (std::size_t i = 0; i < nVertices; ++i) {
    long vertexId = vertexIds[interfaceInfo.getCCWOrderedVertex(i)]
    coordinates[i] = getVertex(vertexId).getCoords();
}
```

or add to PatchKernel a function like "ConstProxyVector<long> VolumeKernel::getFaceOrderedVertexIds(const Cell &cell, int face) const" that takes in input an interface.

Reply:
Done by using the code you provided